aa_stripe/api.py

```diff
@@ -17,11 +19,22 @@ class CouponDetailsAPI(RetrieveAPIView):
     lookup_field = "coupon_id"
 
 
-class CustomersAPI(CreateAPIView):
+class CustomersAPI(CreateAPIView, RetrieveModelMixin):
     queryset = StripeCustomer.objects.all()
-    serializer_class = StripeCustomerSerializer
     permission_classes = (IsAuthenticated,)
 
+    def get_object(self):
+        return get_object_or_404(self.get_queryset(), user=self.request.user)
+
+    def get_serializer_class(self):
```

Comment:
if hasattr(self, "request") - otherwise it will breat swagger - this method may be called w/o request set.

Also, one of the serializers (preferably the POST) should be returned by default (return self.serializer_class)